src/sentry/dynamic_sampling/prioritise_transactions.py

```diff
@@ -109,6 +116,85 @@ def get_orgs_with_project_counts(max_orgs: int, max_projects: int) -> Iterator[L
         yield [org_id for org_id, _ in last_result]
 
 
+def fetch_project_transaction_totals(org_ids: List[int]) -> Iterator[ProjectTransactionsTotals]:
+    """
+    Fetches the total number of transactions and the number of distinct transaction types for each
+    project in the given organisations
+    :param org_ids:
+    :return: an iterator of org_ids
+    """
+    start_time = time.time()
+    offset = 0
+    org_ids = list(org_ids)  # just to be sure it is not some other sequence
+    transaction_string_id = indexer.resolve_shared_org("transaction")
+    transaction_tag = f"tags_raw[{transaction_string_id}]"
+    metric_id = indexer.resolve_shared_org(str(TransactionMRI.COUNT_PER_ROOT_PROJECT.value))
+    more_results = True
+
+    while more_results and (time.time() - start_time) < MAX_SECONDS:
+        query = (
+            Query(
+                match=Entity(EntityKey.GenericOrgMetricsCounters.value),
+                select=[
+                    Function("sum", [Column("value")], "num_transactions"),
+                    Function("uniqExact", [Column(transaction_tag)], "num_classes"),
```

Comment:
@nikhars , thanks for the input, I've seen the documentation and also thought about it. 
The problem I have is that I don't know how inexact is uniq... if the uniq response is different from the real count the overall sampling rate is not going to equal the desired sampling rate ( by how much is a function of how far uniq is from the real number).
It is not clear to me how much of a problem using `uniq` is going to cause.

I would like to try (will ramp it up very slowly) just how taxing is `uniqExact` and if it makes an impact in the task execution we can try using `uniq`. 

The nice thing is the previous implementation didn't have the query at all and I can see if the whole extra query has an impact on the execution time. My expectation is it will not impact it very much since this query should return a lot less data, but I will definitely compare to what we have now.

What do you think ?

Reply:
@nikhars thanks for the context.
1% error (which is what it showed in the example) is acceptable, I will take your advise and switch to `uniq`.

My expectation is that the number of transaction classes in in the tens or hundreds, so given the fact that
we have a small cardinality I didn't expect memory problems, did not really think in terms of speed.
 
Sentry project which has a lot of transaction types has around 750, but most projects I looked at have under 100.

I will change it to `uniq` and try to monitor how far off we are from the target rate, we can never be exact anyways since we always base our rebalance on the transaction distribution of the previous hour and apply it to the transactions in the next hour.

Thanks again for you advise.